db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/WindowResultSetNode.java

```diff
@@ -495,6 +495,35 @@ private void addOverColumns(Map<String, ColumnRefPosition> pulledUp) throws Stan
                         // Cache the ref
                         pulledUp.put(expKey, new ColumnRefPosition(bottomRC, childRCL.size()));
                     }
+                } else {
+                    // Check again if we find match, if not then add it in cache ref if we have the equivalent match
+                    if (pulledUp.get(expKey) == null) {
```

Comment:
no, it won't be always true; in-fact it should be mostly false especially when contains() returns true, it is supposed to find the entry in the map but due to aliases it does not, and so we need to add it with alias.